### ISSUE TITLE
BottleNeck.ipynb  improvement comments

### ISSUE THREAD

**Issue by Arsey** [anhlt/keras-102-flower-dataset#1]
Hi! I've investigated the code in BottleNeck.ipynb and it's helped me to fix my issues. But also I've noticed some things that might be fixed and simplified in your code:
1. you don't need to make original VGG16 layers not trainable as on this stage it does not train and just extract bottleneck features 
```Python
for layer in base_model.layers[:15]:
    layer.trainable = False
```
2. **class_mode=None** doesn't influence on results. Also I would recommend to set classes manually on any stage to prevent some unexpected issues. Previously I had an issue with the last step (fine-tuning) and couldn't fix it for a few weeks, and the problem was in classes and how they were injected into the data generators automatically.
3. you can use **train_labels** without converting them to **train_labels_one_hot**, just change **loss='categorical_crossentropy'** to **loss='sparse_categorical_crossentropy'**
